executor/executor_test.go

```diff
@@ -5949,9 +5949,13 @@ func (s *testSuite) TestKillTableReader(c *C) {
 	tk.MustExec("create table t (a int)")
 	tk.MustExec("insert into t values (1),(2),(3)")
 	c.Assert(failpoint.Enable(retry, `return(true)`), IsNil)
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
 		c.Assert(int(errors.Cause(tk.QueryToErr("select * from t")).(*terror.Error).ToSQLError().Code), Equals, executor.ErrQueryInterrupted.Code())
+		wg.Done()
 	}()
 	time.Sleep(1 * time.Second)
```

Comment:
Could this sleep be removed now?

Reply:
It trys to wait the `select` in the goroutne to run.